plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,565 @@
+            .put("VarbinaryType", WriteMapping.sliceMapping("varbinary", StandardColumnMappings.varbinaryWriteFunction()))
+            .put("DateType", WriteMapping.longMapping("date", snowFlakeDateWriter()))
+            .buildOrThrow();
+
+    private static final Map<String, WriteMappingFunction> SNOWFLAKE_WRITE_MAPPINGS = ImmutableMap.<String, WriteMappingFunction>builder()
+            .put("TimeType", type -> WriteMapping.longMapping("time", SnowflakeClient.snowFlaketimeWriter(type)))
+            .put("ShortTimestampType", SnowflakeClient::snowFlakeTimestampWriter)
+            .put("ShortTimestampWithTimeZoneType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("LongTimestampType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("LongTimestampWithTimeZoneType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("VarcharType", SnowflakeClient::snowFlakeVarCharWriter)
+            .put("CharType", SnowflakeClient::snowFlakeCharWriter)
+            .put("LongDecimalType", SnowflakeClient::snowFlakeDecimalWriter)
+            .put("ShortDecimalType", SnowflakeClient::snowFlakeDecimalWriter)
+            .buildOrThrow();
+
+    @Inject
+    public SnowflakeClient(
+            BaseJdbcConfig config,
+            ConnectionFactory connectionFactory,
+            QueryBuilder queryBuilder,
+            IdentifierMapping identifierMapping,
+            RemoteQueryModifier remoteQueryModifier)
+    {
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, remoteQueryModifier, false);
+
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        ConnectorExpressionRewriter<ParameterizedExpression> connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
+                .addStandardRules(this::quoted)
+                .build();
+
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter<>(
+                connectorExpressionRewriter,
+                ImmutableSet.<AggregateFunctionRule<JdbcExpression, ParameterizedExpression>>builder()
+                        .add(new ImplementCountAll(bigintTypeHandle))
+                        .add(new ImplementCount(bigintTypeHandle))
+                        .add(new ImplementMinMax(false))
+                        .add(new ImplementSum(SnowflakeClient::toTypeHandle))
+                        .add(new ImplementAvgFloatingPoint())
+                        .add(new ImplementAvgDecimal())
+                        .build());
+    }
+
+    @Override
+    public void abortReadConnection(Connection connection, ResultSet resultSet)
+            throws SQLException
+    {
+        // Abort connection before closing. Without this, the Snowflake driver
+        // attempts to drain the connection by reading all the results.
```

Comment:
is this actually true or a copy-paste from MySQL/MariaDB?

Reply:
I think it was a copy-paste from the beginning. I'll remove it. 
@dprophet Let me know if this is something you'd like to keep.